frontend/apps/reader/modules/readersearch.lua

```diff
@@ -256,29 +255,60 @@ function ReaderSearch:onShowSearchDialog(text, direction, regex, case_insensitiv
         -- Keep the LTR order of |< and >|:
         from_start_text, from_end_text = BD.ltr(from_end_text), BD.ltr(from_start_text)
     end
+    self.wait_button = ButtonDialog:new{
+        buttons = {{{ text = "⌛" }}},
+    }
+    local function do_search_edge(func, text, regex, case_insensitive)
+        if regex then
+            return function()
+                self.wait_button.alpha = 0.75
+                UIManager:show(self.wait_button)
```

Comment:
Are you sure it will always be slow? We had quite some discussions and changes and tries with the dict lookup message, some wants it, some others don't...
Where is this button shown on the screen? I guess as it's a ButtonDialog, in the middle, and it will bet overwritten by the search arrow bars, so it shouldn't cause too many refreshes.

Reply:
The button has the same size/position as the search dialog, which is shown right after.

Anyway the button is just shown on regex searches. So if you don't use that you would not see a change :)

The problem with regex is, that we don't have control what the user enters. For searching "xxx|yyyyy|zz" the search is as fast as the normal search. But if a smart guy enters "[a-z]" the time increases (on Tolino) to 10 seconds (3 sec for searching, 7 sec to calculate the highlightings(?) ). In such a case there must be a user notice.